simulator/mips/t/unit_test.cpp

```diff
@@ -141,22 +141,38 @@ TEST_CASE ( "MIPS32_instr: disasm CP1 instructions")
     CHECK(MIPS32Instr(0x460008cb).get_disasm() == "floor.l.s $f3, $f1");
     CHECK(MIPS32Instr(0x462008cf).get_disasm() == "floor.w.d $f3, $f1");
     CHECK(MIPS32Instr(0x460008cf).get_disasm() == "floor.w.s $f3, $f1");
+    CHECK(MIPS32Instr(0xd4230fa3).get_disasm() == "ldc1 $f3, 0xfa3($f1)");
+    CHECK(MIPS32Instr(0xc4230fa3).get_disasm() == "lwc1 $f3, 0xfa3($f1)");
+    CHECK(MIPS32Instr(0x4d2b0041).get_disasm() == "ldxc1 $f1, $t1, $t3");
+    CHECK(MIPS32Instr(0x4d2b0040).get_disasm() == "lwxc1 $f1, $t1, $t3");
     CHECK(MIPS32Instr(0x44090800).get_disasm() == "mfc1 $t1, $f1");
     CHECK(MIPS32Instr(0x462008c6).get_disasm() == "mov.d $f3, $f1");
     CHECK(MIPS32Instr(0x460008c6).get_disasm() == "mov.s $f3, $f1");
+    CHECK(MIPS32Instr(0x462908d3).get_disasm() == "movn.d $f3, $f1, $t1");
+    CHECK(MIPS32Instr(0x460908d3).get_disasm() == "movn.s $f3, $f1, $t1");
+    CHECK(MIPS32Instr(0x462908d2).get_disasm() == "movz.d $f3, $f1, $t1");
+    CHECK(MIPS32Instr(0x460908d2).get_disasm() == "movz.s $f3, $f1, $t1");
     CHECK(MIPS32Instr(0x44890800).get_disasm() == "mtc1 $f1, $t1");
     CHECK(MIPS32Instr(0x462208c2).get_disasm() == "mul.d $f3, $f1, $f2");
     CHECK(MIPS32Instr(0x460208c2).get_disasm() == "mul.s $f3, $f1, $f2");
     CHECK(MIPS32Instr(0x462008c7).get_disasm() == "neg.d $f3, $f1");
     CHECK(MIPS32Instr(0x460008c7).get_disasm() == "neg.s $f3, $f1");
+    CHECK(MIPS32Instr(0x462008d5).get_disasm() == "recip.d $f3, $f1");
+    CHECK(MIPS32Instr(0x460008d5).get_disasm() == "recip.s $f3, $f1");
     CHECK(MIPS32Instr(0x462008c8).get_disasm() == "round.l.d $f3, $f1");
     CHECK(MIPS32Instr(0x460008c8).get_disasm() == "round.l.s $f3, $f1");
     CHECK(MIPS32Instr(0x462008cc).get_disasm() == "round.w.d $f3, $f1");
     CHECK(MIPS32Instr(0x460008cc).get_disasm() == "round.w.s $f3, $f1");
+    CHECK(MIPS32Instr(0x462008d6).get_disasm() == "rsqrt.d $f3, $f1");
+    CHECK(MIPS32Instr(0x460008d6).get_disasm() == "rsqrt.s $f3, $f1");
+    CHECK(MIPS32Instr(0xf4230fa3).get_disasm() == "sdc1 $f3, 0xfa3($f1)");
```

Comment:
Could you please demonstrate what the source of that disassembly is? Floating point register cannot be a source of address.

Reply:
That's a mistake.